experiments/snippets/csvUpload/CSV_Parser.xml

```diff
@@ -0,0 +1,232 @@
+};
+
+if (!inputData) {
+    return reject("No CSV data provided.");
+}
+
+var result = Papa.parse(inputData);
+
+if (result.errors &amp;&amp; result.errors.length) {
+    return reject("Unable to parse CSV File: " + result.errors.join(", "));
+}
+
+if (!result.data.length) {
+    return reject("No data found in CSV file");
+}
+
+var dataModel = skuid.$M("Data");
+var rows = result.data;
+// Fuzzy-match the field names to data
+var columnNames = rows[0];
+var fieldNamesByColumnIndex = [];
+var modelFieldIdsByLabel = {};
+skuid.$.each(dataModel.fields, function(i, field) {
+   modelFieldIdsByLabel[field.label.toLowerCase()] = field.id; 
+});
+var mappedAllColumns = true;
+skuid.$.each(columnNames, function(columnIndex, columnName) {
+    var useFieldName;
+    // Ideal case -- column name matches field name exactly
+    if (dataModel.getField(columnName)) {
+        useFieldName = columnName;
+    } else {
+        // Try lower-cased column name label
+        var lowerCaseColumnName = columnName.toLowerCase();
+        if (modelFieldIdsByLabel[lowerCaseColumnName]) {
+            useFieldName = modelFieldIdsByLabel[lowerCaseColumnName];
+        }
+    }
+    if (!useFieldName) {
+        useFieldName = -1;
+    }
+    fieldNamesByColumnIndex.push(useFieldName);
+});
+
+// Now parse the data and create rows
+$.each(rows, function(i, sourceRow) {
+
+    // Skip first row, which is the column names
+    if (i === 0) return true;
+
+    dataModel.createRow({
+        doAppend: true,
+        additionalConditions: $.map(sourceRow, function(columnValue, columnIndex) {
+            var useFieldName = fieldNamesByColumnIndex[columnIndex];
+            if (useFieldName !== -1) {
+                return {
+                    field: useFieldName,
+                    value: columnValue,
+                };   
+            } else {
+                return null;
+            }
+        }),
+    });
+});
+</jsitem>
+			<jsitem location="staticresource" name="papaparse.min.js" cachelocation="false" url="" content_type="text/javascript">var params = arguments[0],
+	$ = skuid.$;
+</jsitem>
+		</javascript>
+		<actionsequences uniqueid="sk-pU4-134"/>
+		<labels/>
+	</resources>
+	<pageregioncontents>
+		<pageregioncontent uniqueid="__header">
+			<components>
+				<skuid__layoutRegion sticky="true">
+					<components>
+						<skuid__header uniqueid="sk-pUJ-218" model="Input" title="CSV Data Import" subtitle="">
+							<skuid__buttonSet>
+								<groups>
+									<skuid__buttonGroup>
+										<buttons>
+											<skuid__button uniqueid="sk-pUi-321" label="Remove imported data" styleSettingsVariant="default">
+												<actions>
+													<action type="cancel">
+														<models>
+															<model>Data</model>
+														</models>
+													</action>
+												</actions>
+											</skuid__button>
+											<skuid__button uniqueid="sk-ptT-156" label="Download Sample Data" styleSettingsVariant="default">
```

Comment:
Since this was right beside a sentence case button... I had to.
```suggestion
											<skuid__button uniqueid="sk-ptT-156" label="Download sample CSV" styleSettingsVariant="default">
```